File: ts/lcof/21.ts
function exchange(nums: number[]): number[] {
  let l = 0, r = nums.length - 1, tmp
  while (l < r) {
    while (nums[l] % 2 === 1) {
      l += 1
    }
    while (nums[r] % 2 === 0) {
      r -= 1
    }
    if (l < r) {
      tmp = nums[r]
      nums[r--] = nums[l]
      nums[l++] = tmp
    }
  }
  return nums
};

File: py3/lcof/21.py
from typing import List

class Solution:
    def exchange(self, nums: List[int]) -> List[int]:
        l, r = 0, len(nums) - 1
        while l < r:
            if nums[l] % 2:
                l += 1
                continue
            if nums[r] % 2 == 0:
                r -= 1
                continue
            t = nums[r]
            nums[r] = nums[l]
            nums[l] = t

        return nums